### ISSUE TITLE
Unwanted horizontal scrolling in ajaxFiltering

### ISSUE THREAD

**Issue by fortrax** [olifolkerd/tabulator#1381]
Hi Oli,
How are you? Consider the following sample:
```
elem.tabulator({
                    height: "311px",
                    ajaxURL: "/api/sample",
                    ajaxProgressiveLoad: "scroll",
                    ajaxFiltering: true,
                    paginationSize: 20,
                    columns: [
                        { title: "Name", field: "name", sorter: "string", headerFilter: "input", width: 2000 },
                        { title: "Gender", field: "gender", sorter: "string", headerFilter: "input" },
                    ],
                });
```
Note that the "Name" column is configured to be extremely wide. If you run this example and scroll to the right, you can enter something in the header filter of the "Gender" column. After a second or so, Tabulator will call the Ajax API again and display the results. The "Gender" header filter keeps the focus (which is good) but unfortunately, the entire display is scrolled to the exteme left so that the cursor becomes invisible. Even worse, when you type something now, the header line is scrolled (I guess the browser does that to show where you're typing) but the table contents is not so that the layout appears broken. 

Looking forward to 4.0! Thank you so much for what you are doing.

Cheers,
Andreas

**Comment by Rodbjartson** [comment]
@olifolkerd could you please look into this?

**Comment by olifolkerd** [comment]
Hey @fortrax 

Sorry for taking so long to get back to you, i have been deep in v4.0 development.

This issue arises because filtering the data is triggering the table data to be reloaded, which is then resting the table position because it is a full data reload rather than a local filter.

This is consistent behavior with other filter actions but i can see how it is rather undesirable behaviour. I will add it to my list to have a look at

Thanks for bringing it to my attention.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @fortrax 

Try now on the 4.0 branch, the issue should hopefully be resolved.

Cheers

Oli :)

**Comment by fortrax** [comment]
I sure will, but after my holidays. Thank you so much!

**Comment by olifolkerd** [comment]
@fortrax 

Sorry my bad, i got my issues confused there and fixed a different scrolling one. It is fine i know where the issue is and how to solve it, once i have finished getting all the promises in over the weekend it should be sorted.

Cheers

Oli

**Comment by fortrax** [comment]
Hi Oli,

I was kept busy a while on other things. Today I downloaded version 4.1.2 and this issue is unfortunately still around. I live with the following workaround in function Page._parseRemoteData:
 
    if (this.progressiveLoad) {
       ...
    } else {
      ...
       this.table.rowManager.setData(data[this.paginationDataReceivedNames.data], true);

Work like a charm. Maybe this would even be the solution?

Cheers,
Andreas

**Comment by fortrax** [comment]
Hi Oli, any chance to re-open this issue (see above)? Can you even see my comments here? Cheers, Andreas

**Comment by olifolkerd** [comment]
Hey @fortrax 

If you have a working example of the issue that you can share i would be happy to reopen.

Im having to take a slightly harder line on bug reporting as im getting a lot of people raising very vauge reports which just turn out to be misunderstanding on how to use the system, so unless there is a definite example of the issue in action i am starting to have to turn things down.

Cheers

Oli :)

**Comment by fortrax** [comment]
Hey Oli,

I can perfectly understand what you're saying. On the other hand, the ajax scenario cannot be reproduced without a server and then I cannot give you a simple example. If you had a sample ajax server, it would be easier, do you?

The present bug should be common sense, though. Changing a header filter leads to an ajax call, a re-load of the contents but should NOT throw the user out of the filter field. Especially after you've given him or her hard-coded (!) 300 ms time between keystrokes to type the correct value. The fix given above is super-easy.

It seems to me that the whole ajax business is more theory than practice. If the ajax call is slow, the present implementation does not suffice. Every keystroke starts parallel ajax calls but if more than one comes back you write warnings on the console and close the waiting cursor upon the first.

Good, I will keep a growing list of modifications then for updates in the hope that later on you will care more about ajax. Still a fan of Tabulator!

Cheers,
Andreas

**Comment by olifolkerd** [comment]
@fortrax 

The fix you list above is a simplistic work around that fixes your issue but does not deal with the root cause of the issue, and will cause issues with tables configured in different ways. When i apply fixes to Tabulator i have to consider all usage cases.

I take a great deal of offence where you say i dont "care" about the ajax system. I spend 100's of hours each month of my own unpaid time on Tabulator, and recently spent several days completely rebuilding the entire ajax system to make it more usable and add a load of additional features in 4.0 as well as adding a lot of automatic configuration functionality to it in the 4.2 release. So please consider your words next time.

On top of the time i spend developing Tabulator i also answer between 50 - 200 questions a month from users, it can take hours in some cases to manually build a table matching users descriptions, and i simply dont have time to do that for everyone, so i dont think it is out of order to ask developers to spend 5 mins of their own time setting up a demo to better explain their issues. there are many free public servers available for sample ajax requests as well.

Cheers

Oli :)

**Comment by fortrax** [comment]
Oh my god, I said that all wrong! I meant "care" in the sense of "focus on", not "hold dear". Please don't be offenced, it wasn't meant that way! And of course, yes, I was too lazy too think about that probably out in the internet their should be all kinds of test servers. And I will set out to write that example.

You know what - my situation is a bit similar to yours. I want to get rid of angular in our company and therefore I had to replace gridUI with your stuff. I don't regret it a moment but it explains why I am doing this in _my_ free time ;-)

Greatly appreciating what you're doing,
Best wishes,
Andreas

**Comment by fortrax** [comment]
Well it wasn't five minutes but not too long. I created my first jsfiddle. The latest Tabulator on cloudflare seems to be 4.1.5 but 4.2 shows a similar behaviour. I did not even implement the ajax part as the behaviour occurs [without](https://jsfiddle.net/fortrax/5z2un6j4/22/).

Click on "Run", scroll the table to the right, click on the filter of column "last name" and enter something in the header, typing slowly. The pane will scroll the "last name" out of sight under your feet. As soon as 4.2 is available on cloudflare, I'll be happy to update the jsfiddle.

Thank you so much!
Andreas Fordan

**Comment by olifolkerd** [comment]
Hey @fortrax 

All good, meanings can get confused really easily,

Thanks for taking the time to put togeather that js fiddle, it makes things really clear.

**Comment by fortrax** [comment]
Thank you! BTW, I really mean that I can live with a list of work arounds as long as it does not grow too much. So there was and is no hurry or pressure intendend. In time, I will create an issue for other ajax problems (this time having a fiddle ready).

**Comment by olifolkerd** [comment]
Hey Guys,

You will be happy to hear that a fix for this has gong into the 4.6 branch, which will be released later today.

Cheers

Oli :)
